src/network/mod.rs

```diff
@@ -44,7 +57,7 @@ impl interactive_clap::FromCli for Network {
             .expect("Failed to get network config!")
             .clone();
 
-        match (context.on_after_getting_network_callback)(&network_config) {
+        match (context.on_after_getting_network_callback)(&network_config, wallet_url) {
```

Comment:
Instead of adding another parameter to the callback, let's just update the `network_config` when `optional_wallet_url` is Some. This way you won't need to "teach" all the handlers how to react to the `optional_wallet_url`, and they can just rely on `network_config.wallet_url`.

Reply:
Done